tracing/http.go

```diff
@@ -0,0 +1,82 @@
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/reddit/baseplate.go/httpbp"
+	"github.com/reddit/baseplate.go/thriftbp"
+)
+
+// StartSpanFromHTTPContext creates a server span from http context object.
+//
+// This span would usually be used as the span of the whole http endpoint
+// handler, and the parent of the child-spans.
+//
+// Caller should pass in the context object they got from http library,
+// which would have all the required headers already injected.
```

Comment:
I think we should mention here that registering `PopulateRequestContext` in `ServerBefore` is required.

Reply:
The last paragraph of the comment should be merged here. This paragraph works for the thrift case because the feature is built into thrift's `SimpleServer`, no extra steps needed. Here we do need the extra step of `PopulateRequestContext`. Saying "context object they got from http library ... have all the required headers already injected" is kinda untrue :)

Also I think you actually meant go-kit's http library here, which should also be called out (by default people would assume you are talking about stdlib http package, which does not come with contexts in handlers).

Reply:
now we no longer need this paragraph.